### ISSUE TITLE
Are the DNase arrays IF matrices, and why the sample count mismatch?

### ISSUE THREAD

**Issue by HappyLife-together** [OluwadareLab/DiCARN_DNase#2]
Hi ,
I’m trying to understand the DNase files in the repo and how they line up with the Hi-C pipeline.
In dicarn_project_data/DNase/Source/dicarn_10kb40kb_c40_s40_b201_nonpool_train_K562_src.npz, what do the data and target arrays represent?
Are these the IF matrices produced by the DNase-seq processing, or something else?
After running the published Hi-C preprocessing I get tensors of shape (42832, 1, 40, 40), but the DNase file above has shape (95126, 1, 40, 40).
I’d like to know what causes this difference and how to align the two modalities so every Hi-C sample has a matching DNase sample.

Thanks for any pointers!

**Comment by HappyLife-together** [comment]
I can only train DiCARN with Hi-C data; with your code and released data I’m unable to train DiCARN-DNase.

**Comment by Marblegee** [comment]
Hello,
About the IF matrices,`TensorDataset()` as used in `train_set = TensorDataset(combined_train_data, combined_train_target, combined_train_inds)` gives away the needed information. `TensorDataset()` takes in your feature arguments, typically consisting of your features and labels. Hence, the data and target arrays represent the downsampled IF matrices and  ground truth IF matrices, all of which are together in the npz data files. Secondly, please make sure to avoid mismatching when working with the various cell line dataset to prevent tensor size issues.

Secondly, the training script (DiCARN_train.py) defaults to DiCARN_DNase training; lines 78-79 `# Concatenate DNase data with HiC data`
`combined_train_data = torch.cat((train_data, dnase_train_data), dim=0)` `combined_train_target = torch.cat((train_target, dnase_train_target), dim=0)` shows that the training data is a combination of the DNase and regular HiC datasets, resulting in DiCARN_DNase. To achieve the DiCARN standalone concept, you have to train with the HiC data alone, where your `train_set` in line 83 is initialized with `train_data`, etc, not `combined_train_data`, etc. Lastly, the datasets provided in Zenodo are correctly outlined. Note that the default GM12878 training and validation datasets are provided in the `dicarn_project_data/Train and validation/` directory, while the DNase datasets are provided thus: `dicarn_project_data/DNase/<Source OR Target>/<data for cell line of interest>`
